source/mainwindow.cpp

```diff
@@ -236,7 +241,30 @@ void MainWindow::saveAs(QString x) {
   if (currentFile != "" && !image.isNull())
     image.save(currentFile + "." + x, x.toUpper().toLocal8Bit().constData());
 }
-
+void MainWindow::rename() {
+  if(currentFile != "" && !image.isNull()) {
+      bool ok;
+      QFileInfo file(currentFile);
+      QString currentName = file.fileName();
+      QString text = QInputDialog::getText(this, tr("Rename"),
+                                           tr("Name for the file:"), QLineEdit::Normal,
+                                            currentName, &ok);
+      if(ok && !text.isEmpty()) {
+          QString renamedFile = file.absoluteDir().absoluteFilePath(text);
+          QFileInfo checkFile(renamedFile); //Check if file exist in that directory
```

Comment:
If code is good enough to be understood by reading, we may remove the comments. It would look better!